src/rust-crypto/rust-crypto.ts

```diff
@@ -1280,6 +1311,54 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, CryptoEventH
         obj.signatures = Object.fromEntries(sigs.entries());
     }
 
+    /**
+     * Implementation of {@link CryptoApi#restoreKeyBackupWithPassphrase}.
+     */
+    public async restoreKeyBackupWithPassphrase(
+        passphrase: string,
+        opts?: KeyBackupRestoreOpts,
+    ): Promise<KeyBackupRestoreResult> {
+        const backupInfo = await this.backupManager.getServerBackupInfo();
+        if (!backupInfo?.version) {
+            throw new Error("No backup info available");
+        }
+
+        const privateKey = await keyFromAuthData(backupInfo.auth_data, passphrase);
+
+        // Cache the key
+        await this.storeSessionBackupPrivateKey(privateKey, backupInfo.version);
+        return this.restoreKeyBackup(opts);
+    }
+
+    /**
+     * Implementation of {@link CryptoApi#restoreKeyBackup}.
+     */
+    public async restoreKeyBackup(opts?: KeyBackupRestoreOpts): Promise<KeyBackupRestoreResult> {
+        // Get the decryption key from the cache
+        const privateKeyFromCache = await this.getSessionBackupPrivateKey();
+        if (!privateKeyFromCache) throw new Error("No decryption key found in cache");
+
+        // Get the backup version from the cache
+        const backupVersion = await this.getSessionBackupVersion();
+        if (!backupVersion) throw new Error("No backup version found in cache");
```

Comment:
Let's avoid calling olmMachine.getBackupKeys twice (and get rid of getSessionBackupVersion).

```suggestion
        // Get the decryption key from the crypto store
        const backupKeys: RustSdkCryptoJs.BackupKeys = await this.olmMachine.getBackupKeys();
        const {decryptionKey, backupVersion} = backupKey;
        if (!decryptionKey || !backupVersion) throw new Error("No decryption key found in crypto store");
```